pkg/virtctl/imageupload/imageupload.go

```diff
@@ -313,25 +309,51 @@ func (c *command) run(args []string) error {
 		return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
 	}
 
-	pvc, err := getAndValidateUploadPVC(virtClient, namespace, name, noCreate, archiveUpload)
+	pvc, err := getAndValidateUploadPVC(virtClient, namespace, c.name, c.noCreate, c.archiveUpload, c.createPVC)
 	if err != nil {
-		if !(k8serrors.IsNotFound(err) && !noCreate) {
+		if !(k8serrors.IsNotFound(err) && !c.noCreate) {
 			return err
 		}
 
-		if !noCreate && len(size) == 0 {
+		if !c.noCreate && len(c.size) == 0 {
 			return fmt.Errorf("when creating a resource, the size must be specified")
 		}
 
 		var obj metav1.Object
 
-		if createPVC {
-			obj, err = createUploadPVC(virtClient, namespace, name, size, storageClass, accessMode, volumeMode, archiveUpload)
+		if c.createPVC {
+			obj, err = createUploadPVC(
+				virtClient,
+				namespace,
+				c.name,
+				c.size,
+				c.storageClass,
+				c.accessMode,
+				c.volumeMode,
+				c.defaultInstancetype,
```

Comment:
looks like you use more params/flags than what is originally sent to the createUploadPVC function. Can you explain why in the commit desc?

Reply:
Yep will do.